velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -850,6 +849,46 @@ void ExpressionFuzzer::logStats() {
   }
 }
 
+void ExpressionFuzzer::ExprBank::insert(const core::TypedExprPtr& expression) {
+  auto typeString = expression->type()->toString();
+  if (typeToExprsByLevel_.find(typeString) == typeToExprsByLevel_.end()) {
+    typeToExprsByLevel_.insert(
+        {typeString, ExprsIndexedByLevel(maxLevelOfNesting_ + 1)});
+  }
+  auto& expressionsByLevel = typeToExprsByLevel_[typeString];
+  int nestingLevel = getNestedLevel(expression);
```

Comment:
nit: ExprBank::insert() is called in generateExpression() that is a recursive function, so getNestedLevel() will execute on every sub-expression that may have overlap with each other (e.g., foo(c0) and then bar(foo(c0))). Maybe we can let generateExpression() return a pair of TypedExprPtr and the nesting level of the generated expression, and pass this nesting level to the ExprBank::insert(). What do you think?

Reply:
@kagamiori That was part of my initial attempt too but it turned out that a change like that would require changing the full call stack that generates args, all the way from ArgsOverrideFunc to generateArg. Since it would be a decent chunk of change, I am planning to do that as a followup refactor? would that work for you?